tiles-request-api/src/test/java/org/apache/tiles/request/locale/URLApplicationResourceTest.java

```diff
@@ -201,4 +224,36 @@ public void testGetInputStreamWithSpace() throws IOException {
     	assertNotNull(is);
     	is.close();
     }
+
+    @Test
+    public void testLocalProtocol() throws IOException {
+        URL url = new URL("test1://foo/bar.txt");
+        URLApplicationResource resource = new URLApplicationResource("org/apache/tiles/request/test/locale/resource.txt", url);
+        try {
+            resource.getInputStream();
+        } catch (FileNotFoundException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void testAdditionalRemoteProtocolViaSystemProperties() throws Exception {
+        setProperty(REMOTE_PROTOCOLS_PROPERTY, "test1;test2");
+        Field f = URLApplicationResource.class.getDeclaredField("REMOTE_PROTOCOLS");
+        Field m = Field.class.getDeclaredField("modifiers");
+        m.setAccessible(true);
+        m.setInt(f, f.getModifiers() & ~FINAL);
+        f.setAccessible(true);
+        f.set(URLApplicationResource.class, initRemoteProtocols());
+
+        URL url = new URL("test1://foo/bar.txt");
+        URLApplicationResource resource = new URLApplicationResource("org/apache/tiles/request/test/locale/resource.txt", url);
+        try {
+            resource.getInputStream();
+        } catch (FileNotFoundException e) {
+            fail("FileNotFoundException not allowed here");
+        } catch (IOException e) {
+            assertEquals(EXPECTED_MESSAGE, e.getMessage());
```

Comment:
In my opinion, we should catch the IOException here, and, should assert that it was actually thrown by the TestUrlStreamHandler. This should make sure that the test does not accidentally pass even when an unexpected IOException occurs.